PurchasesCoreSwift/Networking/HTTPClient.swift

```diff
@@ -110,6 +110,12 @@ private extension HTTPClient {
                                             retried: retried,
                                             completionHandler: maybeCompletionHandler)
         }
+
+        if serially {
```

Comment:
@aboedo I think this is actually what we want here, I noticed that the HTTPClient tests are a little flaky, so I made 2 changes, this one just potentially fixes a very minor bug where we don't want to use the serial queue if we're not executing serially.